controllers/azurecluster_controller.go

```diff
@@ -157,7 +167,8 @@ func (r *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterSco
 	}
 
 	if azureCluster.Status.Network.APIServerIP.DNSName == "" {
-		clusterScope.Info("Waiting for API server endpoint to exist")
+		clusterScope.Info("Waiting for Load Balancer to exist")
```

Comment:
so I looked into this a bit more, it's actually not accurate right now because the DNSName gets set by the azurecluster reconciler _before_ the load balancer gets created, so it not being empty doesn't indicate that the load balancer exists.

Reply:
see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/652#discussion_r447940198

Reply:
doesn't the create `createOrUpdateNetworkAPIServerIP` happen in the `Reconcile` a few lines before this? ` https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azurecluster_controller.go#L154-L163

Reply:
It does. But in that `Reconcile`, `createOrUpdateNetworkAPIServerIP` is the first func that gets called and `azureCluster.Status.Network.APIServerIP.DNSName` is set before the Load Balancer is created. So it could be that this gets set and later the Load Balancer create fails.

Reply:
I assumed if something fails the call to reconcile from the controller should exit with error and not get to that point in the code https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azurecluster_controller.go#L156

Reply:
Good point, I didn't think of that. In that case I think it's okay 👍